src/shim/runtime/v2/rune/v2/create_test.go

```diff
@@ -0,0 +1,141 @@
+)
+
+const (
+	testDirMode  = os.FileMode(0750)
+	testFileMode = os.FileMode(0640)
+)
+
+//go:embed testdata/busybox.json
+var busyboxConfigJson []byte
+
+func SetupOCIConfigFile(t *testing.T) (rootPath string, bundlePath, ociConfigFile string) {
+	assert := assert.New(t)
+
+	tmpdir := t.TempDir()
+
+	bundlePath = filepath.Join(tmpdir, "bundle")
+	err := os.MkdirAll(bundlePath, testDirMode)
+	assert.NoError(err)
+
+	ociConfigFile = filepath.Join(bundlePath, "config.json")
+	err = os.WriteFile(ociConfigFile, busyboxConfigJson, testFileMode)
+	assert.NoError(err)
+
+	return tmpdir, bundlePath, ociConfigFile
+}
+
+func TestCreatePod(t *testing.T) {
+	assert := assert.New(t)
+
+	_, bundlePath, ociConfigFile := SetupOCIConfigFile(t)
+
+	spec, err := config.LoadSpec(filepath.Join(bundlePath, "config.json"))
+	assert.NoError(err)
+
+	// set expected container type and sandboxID
+	spec.Annotations = make(map[string]string)
+	spec.Annotations[testContainerTypeAnnotation] = testContainerTypeSandbox
+	spec.Annotations[testSandboxIDAnnotation] = testSandboxID
+	spec.Annotations[testSandboxNamespaceAnnotation] = testNamespaceTypeKubesystem
+
+	// rewrite file
+	err = config.SaveSpec(ociConfigFile, spec)
+	assert.NoError(err)
+
+	s := &service{
+		id:         testContainerID,
+		containers: make(map[string]*runc.Container),
+		context:    context.Background(),
+	}
+
+	req := &taskAPI.CreateTaskRequest{
+		ID:       testContainerID,
+		Bundle:   bundlePath,
+		Terminal: true,
+	}
+
+	ctx := namespaces.WithNamespace(context.Background(), "UnitTest")
+	_, err = s.Create(ctx, req)
+
+	expectedErr := "OCI runtime create failed"
+	assert.Error(err)
+	assert.Contains(err.Error(), expectedErr)
+}
+
+func TestCreateApp(t *testing.T) {
+	assert := assert.New(t)
+
+	_, bundlePath, ociConfigFile := SetupOCIConfigFile(t)
+
+	spec, err := config.LoadSpec(filepath.Join(bundlePath, "config.json"))
+	assert.NoError(err)
+
+	// set expected container type, sandboxID, SandboxNamespace and Image
+	spec.Annotations = make(map[string]string)
+	spec.Annotations[testContainerTypeAnnotation] = testContainerTypeContainer
+	spec.Annotations[testSandboxIDAnnotation] = testSandboxID
+	spec.Annotations[testSandboxNamespaceAnnotation] = testNamespaceTypeDefault
+	spec.Annotations[testImageAnnotation] = testImage
+
+	// rewrite file
+	err = config.SaveSpec(ociConfigFile, spec)
+	assert.NoError(err)
+
+	s := &service{
+		id:         testContainerID,
+		containers: make(map[string]*runc.Container),
+		context:    context.Background(),
+		agentID:    testagentID,
+	}
+
+	req := &taskAPI.CreateTaskRequest{
+		ID:       testContainerID,
+		Bundle:   bundlePath,
+		Terminal: true,
+	}
+
+	ctx := namespaces.WithNamespace(context.Background(), "UnitTest")
+	stubs := gostub.Stub(&ContainerBase, t.TempDir())
+	defer stubs.Reset()
+	_, err = s.Create(ctx, req)
+
+	expectedErr := "failed to mount rootfs component"
+	assert.Error(err)
+	assert.Contains(err.Error(), expectedErr)
```

Comment:
use `FailNow()`if these fail since the assertions below are not relevant anymore